### DISCUSSION TITLE
Missing prompt for save before exit?

### DISCUSSION THREAD

**Discussion by willhansen** [neovide/neovide#871] [Q&A] [answered]
In GVim we have a prompt when exiting the window before saving: 

![Question_312](https://user-images.githubusercontent.com/2918280/128926279-0c0095b7-70e7-48f9-874c-a3a439e18df2.png)

In neovide, that seems to not be there.  There's the usual vim exit warning with `:q`, but pressing the window x button can easily lead to lost work.

Is this just my install (v0.7.0), or is this actually not implemented?

**Comment by Kethku** [comment, marked as answer]
This is not implemented. We don't have a way to request to the user whether they want to close or not. One option to fix this would be to change the close command to be :qall rather than :qall!.

**Reply by willhansen** [reply]
It sounds like that would require the user to switch back to the keyboard from the mouse, but I still like it more than no warning at all.

**Reply by rddunphy** [reply]
I'd really appreciate it if this could be changed - or, if some people prefer `qa!`, could a configuration option be added? At the moment it's way too easy to lose work if I accidentally close the wrong window because I've lost track of which one has focus, for instance. This would also be consistent with the behaviour of just about every other text editor out there. (A keyboard-only solution seems fine to me, given how keyboard-oriented Vim is in any case.)

**Reply by viviicat** [reply]
I created a small change #1247 which ought to help rectify this.
